### PR TITLE
NN-1885: Add in amend user email

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: backend/api/oauthApi.js [hunk 1]
```diff
@@ -32,7 +33,7 @@ const oauthApiFactory = (client, { clientId, clientSecret, url }) => {
   const enableUser = (context, { username }) => put(context, `api/authuser/${username}/enable`)
   const disableUser = (context, { username }) => put(context, `api/authuser/${username}/disable`)
   const allRoles = context => get(context, `api/authroles`)
-  const amendUser = (context, username, email) => get(context, `api/authuser/${username}`, email)
+  const amendUser = (context, username, email) => post(context, `api/authuser/${username}`, email)
```
Comment (petergphillips) [inline backend/api/oauthApi.js]: :man_facepalming:

File: src/Admin/MaintainAuthUsers/components/AuthUser.test.js [hunk 2]
```diff
@@ -17,14 +18,16 @@ describe('Auth user display', () => {
...
     const wrapper = renderer.create(
-      <AuthUser
-        contextUser={user}
-        roleList={[]}
-        handleAdd={stubFunc}
-        handleRemove={stubFunc}
-        handleEnable={stubFunc}
-        handleDisable={stubFunc}
-      />
+      <MemoryRouter>
```
Comment (petergphillips) [inline src/Admin/MaintainAuthUsers/components/AuthUser.test.js]: router needed as link using react-router-dom

File: src/Admin/MaintainAuthUsers/containers/AuthUserAmendContainer.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    event.preventDefault()
+
+    if (!validateAmend(this.state, setErrorDispatch)) {
+      return
+    }
+
+    try {
+      await axios.post(
+        '/api/auth-user-amend',
+        {
```
Comment (JamieDraperUK) [inline src/Admin/MaintainAuthUsers/containers/AuthUserAmendContainer.js]: could this just be this.state?

File: src/Admin/MaintainAuthUsers/containers/AuthUserValidation.js [hunk 4]
```diff
@@ -69,4 +76,23 @@ const validateCreate = ({ username, email, firstName, lastName }, setError) => {
   return true
 }
 
-export { validateSearch, validateAdd, validateCreate }
+const validateAmend = ({ email }, setError) => {
```
Comment (JamieDraperUK) [inline src/Admin/MaintainAuthUsers/containers/AuthUserValidation.js]:
```suggestion
const validateAmend = ({ email }, setError) => {
  const errors = []

  if (!email) {
    setError([{ targetName: 'email', text: 'Enter an email address' }])
    return false
  }

  validateEmailFormat(email, errors)

  if (errors.length > 0) {
    setError(errors)
    return false
  }
  return true
}
```

**Pull request merged**